include/spawn/impl/spawn.hpp

```diff
@@ -54,6 +55,45 @@ namespace detail {
     {
     }
 
+    void operator()(Ts... values)
+    {
+      *ec_ = boost::system::error_code();
+      *value_ = std::forward_as_tuple(std::move(values)...);
+      if (--*ready_ == 0)
+        callee_->resume();
+    }
+
+    void operator()(boost::system::error_code ec, Ts... values)
+    {
+      *ec_ = ec;
+      *value_ = std::forward_as_tuple(std::move(values)...);
+      if (--*ready_ == 0)
+        callee_->resume();
+    }
+
+  //private:
+    std::shared_ptr<continuation_context> callee_;
+    continuation_context& caller_;
+    Handler handler_;
+    std::atomic<long>* ready_;
+    boost::system::error_code* ec_;
+    std::tuple<Ts...>* value_;
```

Comment:
We could use std::aligned_storage_t here and then we wouldn't require results to be default constructible.

Reply:
good catch; this was an issue for the single-return-value case also. i wrapped their values in boost::optional<> to avoid the default construction